pkg/query/querier.go

```diff
@@ -360,6 +364,7 @@ func (q *querier) selectFn(ctx context.Context, hints *storage.SelectHints, ms .
 		SkipChunks:              q.skipChunks,
 		Step:                    hints.Step,
 		Range:                   hints.Range,
+		SortWithoutLabels:       replicaLabels,
 	}, resp); err != nil {
```

Comment:
I think this change doubles down on us having a `storepb.SeriesServer` here however how could we achieve full streaming in Select() if it will become pretty much necessary to pass a `storepb.SeriesServer` here? :thinking:

Reply:
I am not sure I fully understand this. If replica labels are external labels, isn't everything going to be streamed? The `sortedSeriesServer` will not buffer if sorting is not required, so I think the effect will be the same no?

Reply:
I think @GiedriusS refers to his idea of streamed series here: https://github.com/fpetkovski/thanos/pull/3

It is true, that no matter how we stream in proxy, we still wait for all responses here: https://github.com/thanos-io/thanos/blob/227928ceb9840808c26f67ae14d863df682f41a6/pkg/query/querier.go#L294 - I though in k-way merge we got rid of this but we did not.

So I would merge this PR and continue our discussion on streaming more in later stages - the problem is that new PromQL engine requires full buffered responses straight away as well. So it does not matter if this is streamed or not unfortunately. WDYT @GiedriusS? I think we have to discuss different StoreAPI (`StoreAPI2` or sth) with some separate request for chunks vs series ideally. That's why we are not prioritizing streaing in proxy now. It does not give us anything - it's better to buffer ALL ASAP for now.

Reply:
Actually we can completely remove `seriesServer` now because `sortedSeriesServer` will buffer if required.

Reply:
:+1: sounds good. Maybe we can clean `seriesServer` up in a follow-up PR to keep this smaller?